components/encryption/src/file_dict_file.rs

```diff
@@ -204,6 +212,15 @@ impl FileDictionaryFile {
         if self.enable_log {
             let file = self.append_file.as_mut().unwrap();
             let bytes = Self::convert_record_to_bytes(name, LogRecord::INSERT(info.clone()))?;
+
+            fail::fail_point!("file_dict_log_insert_error", |_| {
```

Comment:
```suggestion
            fail::fail_point!("file_dict_log_append_incomplete", |_| {
```

Reply:
I presume we can pass in a `bytes_to_write` parameter to this failpoint? Need to test both the header complete and incomplete case.

Reply:
updated, `truncate_num` is added.